src/atlas/projection/detail/CubedSphereProjectionBase.cc

```diff
@@ -327,9 +327,27 @@ void CubedSphereProjectionBase::alphabetat2xy(const idx_t& t, const double ab[],
     double normalisedY = (ab[LAT] + M_PI_4)/M_PI_2;
     xy[XX] = normalisedX * 90. + xOffset[t];
     xy[YY] = normalisedY * 90. + yOffset[t];
+
+    enforceXYdomain(xy);
 }
 
 // -------------------------------------------------------------------------------------------------
+void CubedSphereProjectionBase::enforceXYdomain(double xy[] ) const {
+    // the conversion from lonlat to xy can involve some small errors and small errors
+    // can affect whether xy that is created within a valid space
+    // This has been tested with N = 512 with equiangular and equidistant projections.
+    const double tol{70.0};
+    if (is_same(xy[XX], 0.0, tol)) { xy[XX] = 0.0; }
+    if (is_same(xy[XX], 90.0, tol)) { xy[XX] = 90.0; }
+    if (is_same(xy[XX], 180.0, tol)) { xy[XX] = 180.0; }
+    if (is_same(xy[XX], 270.0, tol)) { xy[XX] = 270.0; }
+    if (is_same(xy[YY], -45.0, tol) && (xy[XX] <= 180.0)) { xy[YY] = -45.0; }
+    if (is_same(xy[YY], 45.0, tol) && (xy[XX] > 180.0)) { xy[YY] = 45.0; }
+    if (is_same(xy[YY], 45.0, tol) && is_same(xy[XX], 0.0)) { xy[YY] = 45.0; }
+    if (xy[YY] >= 135.0) { xy[YY] = 135.0 - __DBL_EPSILON__ ; }
+    if (xy[YY] <= -135.0) { xy[YY] = -135.0 + __DBL_EPSILON__ ; }
+    if (xy[XX] >= 360.0) { xy[XX] = 360.0 + __DBL_EPSILON__ ; }
```

Comment:
Why do you need to add/subtract ```__DBL_EPSILON__``` , and not just snap to value as in
```
xy[YY] = std::min( xy[YY],  135.);
xy[YY] = std::max( xy[YY], -135.);
xy[XX] = std::min( xy[XX],  360.);
xy[XX] = std::max( xy[XX],    0.);
```
You should also use ```std::numeric_limits<double>::epsilon()``` defined in header ```<limits>``` which is equivalent to ```__DBL_EPSILON__```

Reply:
To be well defined I need values in xy to be less than 135 degree in Y more than -135 in Y and less than 360 in X.  For that I need epsilon.  I will rewrite this with min / max.